keps/sig-windows/4802-windows-node-shutdown/README.md

```diff
@@ -0,0 +1,889 @@
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+As specified in [kep2000](https://github.com/zylxjtu/enhancements/tree/master/keps/sig-node/2000-graceful-node-shutdown),
+Kubelet should be aware of node shutdown and trigger graceful shutdown of pods
+during a machine shutdown.
+
+Node graceful shutdown has been available in Linux, this is to implement a comparable feature for Windows node
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Users and cluster administrators expect that pods will adhere to expected pod
+lifecycle including pod termination. Currently, when a node shuts down, pods do
+not follow the expected pod termination lifecycle and are not terminated
+gracefully which can cause issues for some workloads. This KEP aims to address
+this problem by making the kubelet aware of the underlying node shutdown.
+Kubelet will propagate this signal to pods ensuring they can shutdown as
+gracefully as possible.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+*   Focus on handling shutdown on machines where kubelet itself runs as a Windows service, 
+    on machines where kubelet is loaded through other loaders such as nssm.exe, this feature 
+    will not be able to be enabled.
+*   Make kubelet aware of underlying node shutdown event and trigger pod
+    termination with sufficient grace period to shutdown properly
+*   Handle node shutdown in cloud-provider agnostic way
+*   Introduce minimal shutdown delay in order to shutdown node soon as possible
+    (but not sooner)
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+*   Let users modify or change existing pod lifecycle or introduce new inner
+    pod depencides / shutdown ordering
+*   Support kubelet itselft running as a non-Windows service (ex: loaded through nssm.exe), 
+    under those scenarios, kublet will run with this feature being disabled, while operators/developers
+    can get corresponding feature disabling information from the kublet logs
```

Comment:
```suggestion
Support for this feature when the kubelet is not running as a Windows service.
```
Should the kubelet not start if this feature flag is enabled just to make it clear to the operator?

Reply:
There was a discussion on this before, I think we would like to keep it to be similar with the linux way - do not block the kubelet start but have some information in the logs

Reply:
will update lin224